tests/test_logFormatter.py

```diff
@@ -73,31 +83,127 @@ def testButlerLogRecordsFormatter(self):
+            tmp.write(
+                _ButlerLogRecordsModelV1(root=handler.records._records).model_dump_json(
+                    exclude_unset=True, exclude_defaults=True
+                )
+            )
+            tmp.close()
+            ref = DatasetRef(self.datasetType, dataId={}, run=self.run)
+            dataset = FileDataset(path=os.path.abspath(tmp.name), refs=ref)
+            self.butler.ingest(dataset, transfer="move")
 
         records = self.butler.get(ref)
+        self.assertEqual(records, handler.records)
         self.assertEqual(len(records), 2)
 
+    @unittest.skipIf(
+        sys.version_info < (3, 12, 0),
+        "This test requires NamedTemporaryFile behavior not available in older Python versions.",
+    )
+    def testJsonLogRecordsFormatter(self):
+        """Test that externally created JSON format stream files work."""
+        log = logging.getLogger(self.id())
+        log.setLevel(logging.INFO)
+
+        with tempfile.NamedTemporaryFile(mode="w", suffix=".json", prefix="butler-log-") as tmp:
+            handler = FileHandler(tmp.name)
+            handler.setFormatter(JsonLogFormatter())
+            log.addHandler(handler)
+
+            log.info("An INFO message")
+            log.debug("A DEBUG message")
+            log.warning("A WARNING message")
+
+            handler.close()
+
+            # Now ingest the file.
+            ref = DatasetRef(self.datasetType, dataId={}, run=self.run)
+            dataset = FileDataset(path=tmp.name, refs=ref)
+            self.butler.ingest(dataset, transfer="move")
+
+            records = self.butler.get(ref)
+            self.assertEqual(len(records), 2)
+
+    @unittest.skipIf(
+        sys.version_info < (3, 12, 0),
+        "This test requires NamedTemporaryFile behavior not available in older Python versions.",
+    )
+    def testJsonLogRecordsFormatterExtra(self):
+        """Test that externally created JSON format stream files work with
+        extra JSON data appended.
+        """
+        log = logging.getLogger(self.id())
+        log.setLevel(logging.INFO)
+
+        with tempfile.NamedTemporaryFile(mode="w", suffix=".json", prefix="butler-log-") as tmp:
+            handler = FileHandler(tmp.name)
+            handler.setFormatter(JsonLogFormatter())
+            log.addHandler(handler)
+
+            log.info("An INFO message")
+            log.debug("A DEBUG message")
+            log.warning("A WARNING message")
+
+            handler.close()
+            extra_data = {"extra1": 1, "extra2": 2}
+            with open(tmp.name, "a") as stream:
+                ButlerLogRecords.write_streaming_extra(stream, json.dumps(extra_data))
+
+            # Now ingest the file.
+            ref = DatasetRef(self.datasetType, dataId={}, run=self.run)
+            dataset = FileDataset(path=tmp.name, refs=ref)
+            self.butler.ingest(dataset, transfer="move")
+
+            records: ButlerLogRecords = self.butler.get(ref)
```

Comment:
Did you mean to add the type annotation in a test?  Not seeing it in other tests and won't the function be skipped for type checking because it doesn't have type annotations?